### PR TITLE
[tests] fixed codestyle, removed unused code and added several new checks

### PR DESCRIPTION
**Pull request by StrikerRUS**

### CODE REVIEW

**Comment by henry0312** [comment]
How about using [pytest-codestyle](https://pypi.org/project/pytest-codestyle/) and [pytest-isort ](https://pypi.org/project/pytest-isort/)?
I think the tests will be a bit more simple.
Also, [pydocstyle](https://pypi.org/project/pydocstyle/) ([pytest-docstyle](https://pypi.org/project/pytest-docstyle/)) will be good choice.

**Comment by StrikerRUS** [comment]
@henry0312 I like your idea! To be honest, I've already thought about adding this, but haven't found the time.

Will you create a PR?

File: tests/python_package_test/test_sklearn.py [hunk 1]
```diff
@@ -60,20 +60,28 @@ def test_multiclass(self):
...
+                                          '../../examples/lambdarank/rank.train.query'))
+        q_test = np.loadtxt(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                         '../../examples/lambdarank/rank.test.query'))
         gbm = lgb.LGBMRanker()
         gbm.fit(X_train, y_train, group=q_train, eval_set=[(X_test, y_test)],
                 eval_group=[q_test], eval_at=[1, 3], early_stopping_rounds=5, verbose=False,
                 callbacks=[lgb.reset_parameter(learning_rate=lambda x: 0.95 ** x * 0.1)])
+        self.assertLessEqual(gbm.best_iteration_, 15)
+        self.assertGreater(gbm.best_score_['valid_0']['ndcg@1'], 0.5)
+        self.assertGreater(gbm.best_score_['valid_0']['ndcg@3'], 0.5)
```
[the remark this answers is not in the document]
Comment (guolinke) [reply tests/python_package_test/test_sklearn.py]: seems strange, can changing the objective/metric solve this ?

[the remark this answers is not in the document]
Comment (StrikerRUS) [reply tests/python_package_test/test_sklearn.py]: I think that #1664 (1 problem) causes different `best_iteration_`, which in turn affects on such big difference in `best_score_`.

File: tests/python_package_test/test_sklearn.py [hunk 2]
```diff
@@ -57,20 +57,28 @@ def test_multiclass(self):
...
+                                          '../../examples/lambdarank/rank.train.query'))
+        q_test = np.loadtxt(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                         '../../examples/lambdarank/rank.test.query'))
         gbm = lgb.LGBMRanker()
         gbm.fit(X_train, y_train, group=q_train, eval_set=[(X_test, y_test)],
                 eval_group=[q_test], eval_at=[1, 3], early_stopping_rounds=5, verbose=False,
                 callbacks=[lgb.reset_parameter(learning_rate=lambda x: 0.95 ** x * 0.1)])
+        self.assertLessEqual(gbm.best_iteration_, 12)
+        self.assertGreater(gbm.best_score_['valid_0']['ndcg@1'], 0.65)
+        self.assertGreater(gbm.best_score_['valid_0']['ndcg@3'], 0.65)
```
Comment (StrikerRUS) [inline tests/python_package_test/test_sklearn.py]: These checks are passed only on Windows and macOS with Apple Clang.

Comment (guolinke) [reply tests/python_package_test/test_sklearn.py]: did you merge from master?

Comment (StrikerRUS) [reply tests/python_package_test/test_sklearn.py]: Yeah, 2 commits - just recent docs improvements.

image

Comment (guolinke) [reply tests/python_package_test/test_sklearn.py]: this problem should be fixed in #1739
